packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelConfigurationController.java

```diff
@@ -342,6 +350,58 @@ public ResponseEntity<ResponseDeleted> deleteModelConfiguration(
 		}
 	}
 
+	private static void updateModelParameters(
+			final List<ModelParameter> modelParameters, final List<ParameterSemantic> configParameters) {
+		// Create a map from ConfigParameter IDs to ConfigParameter objects
+		final Map<String, ParameterSemantic> configParameterMap = new HashMap<>();
+		for (final ParameterSemantic configParameter : configParameters) {
+			configParameterMap.put(configParameter.getReferenceId(), configParameter);
+		}
+
+		// Iterate through the list of ModelParameter objects
+		for (final ModelParameter modelParameter : modelParameters) {
+			// Look up the corresponding ConfigParameter in the map
+			final ParameterSemantic matchingConfigParameter = configParameterMap.get(modelParameter.getId());
+			if (matchingConfigParameter != null) {
+				// I'm not sure if we want the set the id, name, description here since we have that on model already
```

Comment:
I don't think you want to set name/description. In fact one might question why you'd have that in the data structure ...

Reply:
we're getting rid of it those fields, wrote this last week before those changes were made